src/client/components/chart-data-buttons/chartbar-buttons/chart-data-button.component.js

```diff
@@ -0,0 +1,33 @@
+import React, { useState } from 'react';
+import './chart-data-button.style.css';
+import PropTypes from 'prop-types';
+import Button from '../../button/button.component';
+
+const ChartbarMenu = ({ buttonLabel }) => {
+  const [selectedButton, setSelectedButton] = useState(buttonLabel.buttonWeek);
+  return (
+    <div className="chartbar-button-wrapper">
+      {buttonLabel.map((obj) => {
+        return (
+          <Button
+            type="toggle"
+            size="large"
+            onClick={(e) => {
+              e.preventDefault();
+              setSelectedButton(obj.buttonId);
+            }}
+            toggled={selectedButton === obj.buttonId}
+          >
+            {obj.buttonLabel}
+          </Button>
+        );
+      })}
+    </div>
+  );
+};
+
+ChartbarMenu.propTypes = {
+  buttonLabel: PropTypes.instanceOf(Array).isRequired,
+};
+
+export default ChartbarMenu;
```

Comment:
It looks great! one last comment 🤓 

I would expect `buttonLabel` to be a string but I see that it is not. I would consider passing the following props
`defaultSelection` = which is optional and can be set as default state
`buttons` = which is the array of objects you map through to display all the buttons

Then change the `propTypes` accordingly 👍